allennlp/modules/token_embedder.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict
+
+import torch
+
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+
+class TokenEmbedder(torch.nn.Module):
+    """
+    A ``TokenEmbedder`` is a ``Module`` that takes as input the :class:`~allennlp.data.DataArray`
+    produced by a :class:`~allennlp.data.fields.TextField` and returns as output an embedded
+    representation of the tokens in that field.
+
+    The ``DataArrays`` produced by ``TextFields`` are `dictionaries` with named representations,
+    like "words" and "characters".  When you create a ``TextField``, you pass in a dictionary of
+    :class:`~allennlp.data.TokenIndexer` objects, telling the field how exactly the tokens in the
+    field should be represented.  This class changes the type signature of ``Module.forward``,
+    restricting ``TokenEmbedders`` to take inputs corresponding to a single ``TextField``.  We also
+    add a method to the basic ``Module`` API: :func:`get_output_dim()`.  You might need this if you
+    want to construct a ``Linear`` layer using the output of this embedder, for instance.
+    """
+    def forward(self,  # pylint: disable=arguments-differ
+                text_field_input: Dict[str, torch.Tensor]) -> torch.Tensor:
+        raise NotImplementedError
+
+    def get_output_dim(self) -> int:
```

Comment:
I did the same thing for the encoders, again because of this: https://github.com/allenai/allennlp/pull/34/files#diff-61e4bd71e206332517f4471e9120f3f1R49.